### PR TITLE
💚✅ [build.sh] implement extended tests

### PR DESCRIPTION
**Pull request by jxn-30**
this implements new tests for `build.sh`:
* run within a git repository and outside of a git repository

also, the file is tested on the folowing new operating systems:
* macOS 11
* macOS 12
=> required fixes have been applied

also, the script accepts a new option `--debug` which will print out the current statement to ease debugging.

### CODE REVIEW

File: build/build.sh [hunk 1]
```diff
@@ -115,51 +136,63 @@ if [[ $_RUN_STEP_NODE = true ]]; then
     fi
     [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
     nvm install "$NODE_VERSION"
-    end_time=$(date +%s%N)
-    echo "=== [⬆️] Setup Node.js: $(((end_time - start_time) / 1000000))ms ==="
+    # re-enable debugging output
+    enable_debugging
```
Comment (Cr4zyc4k3) [inline build/build.sh:140]: Why enable printing and disable it without any output?

Comment (jxn-30) [reply build/build.sh:140]: the enabling and disabling at the beginning and end of each step is added automatically. As I don't want to spam the console too much, I've manually disabled the logging.

I will improve the build script to remove these unnecessary enabling/disabling commands.

Comment (jxn-30) [reply build/build.sh]: this is resolved. I improved the script generation :)

**Review by Cr4zyc4k3 — APPROVED** [review]

File: build/build.sh [hunk 2]
```diff
@@ -115,51 +133,60 @@ if [[ $_RUN_STEP_NODE = true ]]; then
     fi
     [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
     nvm install "$NODE_VERSION"
-    end_time=$(date +%s%N)
-    echo "=== [⬆️] Setup Node.js: $(((end_time - start_time) / 1000000))ms ==="
+    end_time=$(now)
+    echo "=== [⬆️] Setup Node.js: $(((10#$end_time - 10#$start_time) / 1000000))ms ==="
```
[the remark this answers is not in the document]
Comment (jxn-30) [reply build/build.sh:137]: ensure that the string is interpreted as a number in decimal format

File: build/build.sh [hunk 3]
```diff
@@ -4,6 +4,24 @@
...
+
+disable_debugging () {
+    if [[ $DEBUG = true ]]; then
+        set +x
+    fi
+}
+
+now () {
+    local timestamp
+    timestamp="$(date +%s%N)"
```
[the remark this answers is not in the document]
Comment (jxn-30) [reply build/build.sh:21]: we use this to get the timestamp to calculate a difference and not show it anywhere.

File: build/build.sh [hunk 4]
```diff
@@ -216,86 +249,93 @@ if [[ $_RUN_STEP_ESLINT = true ]]; then
...
+    start_time=$(now)
     echo "### [ℹ️] git diff ###"
+    enable_debugging
     git --no-pager diff --color-words
-    end_time=$(date +%s%N)
-    echo "=== [ℹ️] git diff: $(((end_time - start_time) / 1000000))ms ==="
-fi
-
-# [📦] zip files
-if [[ $_RUN_STEP_ZIP = true ]]; then
```
[the remark this answers is not in the document]
Comment (jxn-30) [reply build/build.sh:290]: we don't need automatic zipping for full local builds plus we would have to differentiate between linux and mac.

**Pull request merged**